Adding support for string table lookups in dropdown field labels

AnmAtAnm opened the pull request:
Adds `Blockly.utils.replaceMessageReferences()` method to handle string replacement without interpolation tokens.  Effectively uses the same old code, now moved into `tokenizeInterpolation_()`, which takes a parseInterpolationTokens option.

Replaces the direct JavaScript references (not pure JSON, and thus not portable).

Demonstrating this behavior in the `logic_boolean` dropdown.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/google/blockly/818)
<!-- Reviewable:end -->

rachel-fenichel commented on core/utils.js:
```diff
@@ -392,11 +392,38 @@ Blockly.utils.commonWordSuffix = function(array, opt_shortest) {
+}
+
+/**
+ * Parse a string with and replace string table references. For example,
```
I can't parse this comment.  Did you miss a word?

rachel-fenichel commented on core/utils.js:
```diff
@@ -392,11 +392,38 @@ Blockly.utils.commonWordSuffix = function(array, opt_shortest) {
+ *     tokens.
+ * @param {boolean} parseInterpolationTokens Option to parse numeric
+ *     interpolation tokens (%1, %2, ...) when true.
+ * @return {!Array.<string|number>} Array of strings and numbers.
```
@private

RoboErikG commented:
Aside from Rachel's comments LGTM.

AnmAtAnm commented:
Review status: 0 of 4 files reviewed at latest revision, 2 unresolved discussions.

---

*[core/utils.js, line 405 at r1](https://reviewable.io:443/reviews/google/blockly/818#-KaEqdrKtNPpIVvGze47-r1-405:-KaEsSYXXdHPT6Fw33YB:bp0ozkv) ([raw file](https://github.com/google/blockly/blob/4bdaee0e3b2abfaf04e9b6ad3d1c0acdc38ed848/core/utils.js#L405)):*
<details><summary><i>Previously, rachel-fenichel (Rachel Fenichel) wrote…</i></summary><blockquote>

I can't parse this comment.  Did you miss a word?
</blockquote></details>

Corrected and rewrote. 

---

*[core/utils.js, line 424 at r1](https://reviewable.io:443/reviews/google/blockly/818#-KaEqdrKtNPpIVvGze47-r1-424:-KaEqf7a33TuOXDgNJHj:b-896fix) ([raw file](https://github.com/google/blockly/blob/4bdaee0e3b2abfaf04e9b6ad3d1c0acdc38ed848/core/utils.js#L424)):*
<details><summary><i>Previously, rachel-fenichel (Rachel Fenichel) wrote…</i></summary><blockquote>

@private
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly/818)*
<!-- Sent from Reviewable.io -->

rachel-fenichel commented:
LGTM

timdawborn commented:
This PR broke image support in dropdowns. See #848.
